src/oops/SOLID/openClosePrinciple/before/taxes/TaxCalculatorFactory.java

```diff
@@ -0,0 +1,23 @@
+package oops.SOLID.openClosePrinciple.before.taxes;
+
+import oops.SOLID.openClosePrinciple.before.employees.Employee;
+import oops.SOLID.openClosePrinciple.before.employees.FullTimeEmployee;
+import oops.SOLID.openClosePrinciple.before.employees.Intern;
+import oops.SOLID.openClosePrinciple.before.employees.PartTimeEmployee;
+
+public class TaxCalculatorFactory {
+	
+	public static TaxPayable getEmployeeTaxClass(Employee e) {
+		if (e instanceof FullTimeEmployee) {
```

Comment:
Use HashMap instead of instanceOf as discussed in class

Reply:
I had few doubts with that approach. so went with this one. let's discuss the hashmap one.